From 03088d088b45005d71775bb1462ce8d655c74411 Mon Sep 17 00:00:00 2001
From: tropicrainforest <tropicrainforest27017@gmail.com>
Date: Tue, 9 Jun 2026 00:15:41 +0200
Subject: [PATCH 28/28] SConstruct: use the conda-activated MSVC
 env instead of a hardcoded VS edition

mongo 7.0 hardcodes VSINSTALLDIR=.../2022/Professional and defaults MSVC_USE_SCRIPT
to that edition's vcvarsall.bat (it also pins MSVC_TOOLSET_VERSION=14.31). conda-forge
runners on the other hand runs on MSVC 14.4x, so SCons raises MSVCScriptNotFound at
configure.
---
 SConstruct | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index 1cada324..39d1c1cb 100644
--- a/SConstruct
+++ b/SConstruct
@@ -1273,7 +1273,7 @@ def get_windows_sdk_libdir(sdk_version):
 
 
 env_vars.Add('MSVC_USE_SCRIPT', help='Sets the script used to setup Visual Studio.',
-             default=f'{VSINSTALLDIR}/VC/Auxiliary/Build/vcvarsall.bat')
+             default=False)
 
 env_vars.Add('VSINSTALLDIR', help='The path where Visual Studio is installed.',
              default=VSINSTALLDIR)
